cmd/ops_agent_uap_plugin/service_linux.go

```diff
@@ -0,0 +1,201 @@
+
+func runCommand(cmd *exec.Cmd) (string, error) {
+	if cmd == nil {
+		return "", nil
+	}
+	cmd.SysProcAttr = &syscall.SysProcAttr{
+		Pdeathsig: syscall.SIGKILL,
+	}
+	log.Printf("Running command: %s", cmd.Args)
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		log.Printf("Command %s failed, \ncommand output: %s\ncommand error: %s", cmd.Args, string(out), err)
+		return string(out), err
+	}
+	return string(out), nil
+}
+
+func validateOpsAgentConfig(ctx context.Context, runCommand RunCommandFunc, pluginBaseLocation string) error {
+	configValidationCmd := exec.CommandContext(ctx,
+		pluginBaseLocation+"/"+ConfGeneratorBinary,
+		"-in", OpsAgentConfigLocationLinux,
+	)
+	if output, err := runCommand(configValidationCmd); err != nil {
+		return fmt.Errorf("failed to validate the Ops Agent config:\ncommand output: %s\ncommand error: %s", output, err)
+	}
+	return nil
+}
+
+func generateSubagentConfigs(ctx context.Context, runCommand RunCommandFunc, pluginBaseLocation string) error {
+	confGeneratorBinaryFullPath := fmt.Sprintf("%s/%s", pluginBaseLocation, ConfGeneratorBinary)
+	otelConfigGenerationCmd := exec.CommandContext(ctx,
+		confGeneratorBinaryFullPath,
+		"-service", "otel",
+		"-in", OpsAgentConfigLocationLinux,
+		"-out", pluginBaseLocation+"/"+OtelRuntimeDirectory,
+		"-logs", pluginBaseLocation+"/"+LogsDirectory)
+
+	if output, err := runCommand(otelConfigGenerationCmd); err != nil {
+		return fmt.Errorf("failed to generate Otel config:\ncommand output: %s\ncommand error: %s", output, err)
+	}
+
+	fluentBitConfigGenerationCmd := exec.CommandContext(ctx,
+		confGeneratorBinaryFullPath,
+		"-service", "fluentbit",
+		"-in", OpsAgentConfigLocationLinux,
+		"-out", pluginBaseLocation+"/"+FluentBitRuntimeDirectory,
+		"-logs", pluginBaseLocation+"/"+LogsDirectory, "-state", pluginBaseLocation+"/"+FluentBitStateDiectory)
+
+	if output, err := runCommand(fluentBitConfigGenerationCmd); err != nil {
+		return fmt.Errorf("failed to generate Fluntbit config:\ncommand output: %s\ncommand error: %s", output, err)
+	}
+	return nil
+}
+
+func findPreExistentAgents(ctx context.Context, runCommand RunCommandFunc, agentSystemdServiceNames []string) (bool, error) {
+	cmdArgs := []string{"systemctl", "list-unit-files"}
+	cmdArgs = append(cmdArgs, agentSystemdServiceNames...)
+	findOpsAgentCmd := exec.CommandContext(ctx,
+		cmdArgs[0], cmdArgs[1:]...,
+	)
+	output, err := runCommand(findOpsAgentCmd)
+	if strings.Contains(output, "0 unit files listed.") {
+		return false, nil
+	}
+	if err != nil {
+		return false, fmt.Errorf("unable to verify the existing ops agent and legacy agent installations, error: %s", err)
+	}
+	alreadyInstalledAgents := AgentServiceNameRegex.FindAllString(output, -1)
+	log.Printf("The following systemd services are already installed on the VM: %v", alreadyInstalledAgents)
+	return true, fmt.Errorf("conflicting installations identified: %v, please uninstall them before re-starting the Ops Agent Plugin", alreadyInstalledAgents)
```

Comment:
Does restarting make sense here? I understood that the control flow was dictated by the Plugin Manager.

nit: re-starting/restarting